pkg/segment/aggregations/timechartagg.go

```diff
@@ -262,32 +264,49 @@ func SortTimechartRes(timechart *structs.TimechartExpr, results *[]*structs.Buck
 	}
 
 	sort.Slice(*results, func(i, j int) bool {
-		bucketKey1, ok := (*results)[i].BucketKey.(string)
-		if !ok {
-			log.Errorf("SortTimechartRes: cannot convert bucketKey to string: %v", (*results)[i].BucketKey)
+		timestamp1, err1 := extractTimestamp((*results)[i].BucketKey)
+		if err1 != nil {
+			log.Errorf("SortTimechartRes: bucketKey is invalid for index %d: %v", i, err1)
 			return false
 		}
-
-		bucketKey2, ok := (*results)[j].BucketKey.(string)
-		if !ok {
-			log.Errorf("SortTimechartRes: cannot convert bucketKey to string: %v", (*results)[j].BucketKey)
+		timestamp2, err2 := extractTimestamp((*results)[j].BucketKey)
+		if err2 != nil {
+			log.Errorf("SortTimechartRes: bucketKey is invalid for index %d: %v", j, err2)
```

Comment:
nit: I think you could have just used `err` twice instead of `err1` and `err2`, but that's pretty minor, so this is fine